stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -793,6 +793,36 @@ function dot(x::AbstractArray, y::AbstractArray)
     s
 end
 
+"""
+    dot(x, A, y)
+
+Compute the generalized dot product `dot(A'x, y)` between two vectors `x` and `y`,
+without storing the intermediate result of `A'x`. As for the two-argument
+[`dot(_,_)`](@ref), this acts recursively. Moreover, for complex vectors, the
+first vector is conjugated.
+"""
```

Comment:
This docstring should presumably be attached to `function dot end` and not to the `Number` method.

Also, it seems a bit weird to me to define it as `dot(A'x, y)` rather than `dot(x, A*y)`.    The fact that these two *must* be equivalent is the *definition* of [the adjoint `A'`](https://en.wikipedia.org/wiki/Hermitian_adjoint), so it seems safe to me to use the simpler definition `dot(x, A*y)`.

If you have a non-associative `*` algebra, it seems like you would need to define `A'` such that  `dot(A'x, y) == dot(x, A*y)` still holds.

Reply:
Yes, of course. I was stretching things a bit. What I wanted to emphasize is the order of multiplication in the implementation. I’ll change the docstring accordingly.